lib/Net/Stripe.pm

```diff
@@ -1545,10 +1549,70 @@ method _post(Str $path, $obj?) {
     return $self->_make_request($req);
 }
 
+sub BUILD {
+    my ( $self, $args ) = @_;
+
+    my $message = "The API version %s (%s) is not supported by this version of Net::Stripe. " .
+    "This version only supports %s or before. " .
+    "Please download a version-appropriate branch at https://github.com/lukec/stripe-perl/branches.";
+
+    if (defined($self->api_version)) {
```

Comment:
This if-else will become just an "if" when comment (A) is acted on. Noting that a bit of thought will be required to decide on "requested" vs "for your Stripe account" - or whether the difference is worth the effort.